src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -918,6 +918,32 @@ protected override void Generate(DropIndexOperation operation, [CanBeNull] IMode
             builder.EndCommand();
         }
 
+        protected override void Generate(CreateSequenceOperation operation, IModel model, MigrationCommandListBuilder builder)
```

Comment:
Wouldn't it be better to simply call `base.Generate()` with a `CreateSequenceOperation` that has ClrType overridden to long? This would prevent the duplication of that code in this function...

(@austindrenski it's better to wait for me to review too, I know I'm a bit busy these days but still...)

Reply:
@roji Apologies on moving forward with the merge before your review. I'll open a follow-up PR to de-duplicate where possible.

Reply:
@roji How would you suggest to clone `CreateSequenceOperation operation` class with all it's properties? Or you suggest modify its ClrType property "in-place" and return back to original value after calling base.Generate() method?

Reply:
Changing in-place is a possibility, yes. I'm not even sure it's necessary to return back to the original value as these operations are only used for SQL generation - but for correctness's sake it's a good idea.

You could also copy properties one-by-one to a new instance, including annotations, but that would be tedious and brittle in case something gets added in the future.